### PR TITLE
Use `BlueprintZoneConfig` in RSS service plan

### PR DESCRIPTION
**Pull request by andrewjstone**
In anticipation of adding more `BlueprintZoneConfig` variants with more auxiliary information, we stop converting from `OmicronZoneConfig` to `BlueprintZoneConfig` which is not going to be feasible for much longer.

Instead we change the one production code place we do this, RSS, to directly construct `BlueprintZoneConfig` structs rather than do the conversion.

This has some ripple effects, and results in a new persistent v4 sled service plan.

There is one test that still does this conversion, but the function that does it is now moved into that test module and commented heavily. We hope to remove it shortly.

### CODE REVIEW

**Comment by andrewjstone** [comment]
This is not as daunting as it looks. 1000 lines of it is a generated json file.

File: sled-agent/src/rack_setup/plan/service.rs [hunk 1]
```diff
@@ -458,16 +541,19 @@ impl Plan {
             let dataset_name =
                 sled.alloc_dataset_from_u2s(DatasetType::CockroachDb)?;
             let filesystem_pool = Some(dataset_name.pool().clone());
-            sled.request.zones.push(OmicronZoneConfig {
+            sled.request.zones.push(BlueprintZoneConfig {
+                disposition: BlueprintZoneDisposition::InService,
                 // TODO-cleanup use TypedUuid everywhere
```
Comment (jgallagher) [inline sled-agent/src/rack_setup/plan/service.rs:546]: can delete this comment :tada:

**Review by jgallagher — APPROVED** [review]
Looks great; thanks for cleaning up this mess I left behind.

Out of an abundance of paranoia, have you run through RSS on this branch on london or madrid? (Does a4x2 go through this same code path? I forget when it converges back with normal setup.)

**Comment by andrewjstone** [comment]
Thanks for the review @jgallagher. Much appreciated.

> Looks great; thanks for cleaning up this mess I left behind.
> 
> Out of an abundance of paranoia, have you run through RSS on this branch on london or madrid? (Does a4x2 go through this same code path? I forget when it converges back with normal setup.)

I did not do that yet. I'll definitely do it before merge. I think the deploy task runs through this, but I'll run it on a4x2, which definitely runs RSS. It just does it from a config file I believe. I always wait for `Handoff to Nexus complete`, but I'll also double check after a run.

File: nexus/types/src/deployment.rs [hunk 2]
```diff
@@ -616,167 +606,7 @@ pub struct BlueprintZoneConfig {
...
-        Ok(Self {
-            disposition,
-            id: OmicronZoneUuid::from_untyped_uuid(config.id),
-            underlay_address: config.underlay_address,
-            filesystem_pool: config.filesystem_pool,
-            zone_type,
-        })
-    }
-}
+impl BlueprintZoneConfig {}
```
[the remark this answers is not in the document]
Comment (andrewjstone) [reply nexus/types/src/deployment.rs]: Surprised I missed this.

**Comment by andrewjstone** [comment]
> Thanks for the review @jgallagher. Much appreciated.
> 
> > Looks great; thanks for cleaning up this mess I left behind.
> > Out of an abundance of paranoia, have you run through RSS on this branch on london or madrid? (Does a4x2 go through this same code path? I forget when it converges back with normal setup.)
> 
> I did not do that yet. I'll definitely do it before merge. I think the deploy task runs through this, but I'll run it on a4x2, which definitely runs RSS. It just does it from a config file I believe. I always wait for `Handoff to Nexus complete`, but I'll also double check after a run to ensure the new plan gets written.

I tested on a4x2 and Handoff completed successfully. I see the new `rss-service-plan-v4.json` and it contains an `external_ip_id` which only exists in the `BlueprintZoneConfig`.

**Pull request merged**